File: frontend/assets/js/whatsapp.js
export function limparTelefone(telefone) {
  return String(telefone || "").replace(/\D/g, "");
}

export function normalizarTelefoneBrasil(telefone) {
  const limpo = limparTelefone(telefone);

  if (limpo.startsWith("55")) {
    return limpo;
  }

  return `55${limpo}`;
}

export function criarLinkWhatsApp(telefone, mensagem) {
  const telefoneNormalizado = normalizarTelefoneBrasil(telefone);
  return `https://wa.me/${telefoneNormalizado}?text=${encodeURIComponent(mensagem)}`;
}

File: backend/app/util/whatsapp.py
from urllib.parse import quote

from app.util.telefone import normalizar_telefone_brasil


def criar_link_whatsapp(telefone: str, mensagem: str) -> str:
    telefone_normalizado = normalizar_telefone_brasil(telefone)
    return f"https://wa.me/{telefone_normalizado}?text={quote(mensagem)}"
